tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+        complete = (data) -> 'complete'
+
+        spyOn(ws._messagesBuffer, 'push').and.callThrough()
+        promise = ws.send(message, {success, error, complete})
+        expect(ws._messagesBuffer.push).toHaveBeenCalled()#With {message, options, promise}
+        
+
+      it 'should call custom message if strategy chosen is "custom"', (done) ->
+        
+        newOptions =
+          onDisconnect: (that, messageBuffer, reallyError) ->
+            expect(that).toEqual new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+            expect(messageBuffer).toBe ws._messagesBuffer
+            expect(reallyError).toEqual ReallyError
+            done()
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+
+        message = protocol.createMessage('/users')
+        
+        success = (data) -> 'success'
+        error = (reason) -> 'error'
+        complete = (data) -> 'complete'
+
+        ws.send(message, {success, error, complete})
+
+      it 'should fail messages if strategy chosen is "fail" by rejecting the promise', ->
+        newOptions =
+          onDisconnect: 'fail'
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+
+        message = protocol.createMessage('/users')
+        
+        success = (data) -> 'success'
+        error = (reason) -> 'error'
+        complete = (data) -> 'complete'
+
+        spyOn(ws._messagesBuffer, 'push').and.callThrough()
+        promise = ws.send(message, {success, error, complete})
+        promise.catch (e) ->
+          expect(e).toEqual new ReallyError('Connection to the server is not established')
+
+      it 'should return a promise when channel is not connected with any strategy', ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        message = protocol.createMessage('/users')
+        promise = ws.send(message, {})
+        expect(typeof promise.then is 'function').toBeTruthy()
+
+
+  describe 'reconnect', ->
+    it 'should increase number of attempts with one', (done) ->
+      ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+      ws.connect()
+      ws.socket.addEventListener 'open', () ->
+        ws.socket.close()
+      ws.socket.addEventListener 'close', () ->
+        expect(ws.attempts).toEqual 1
+        ws.disconnect()
+        done()
+
+    it 'should reconnect when time out', (done) ->
+      jasmine.clock().install()
+      ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+      ws.connect()
+      spyOn(ws, 'reconnect').and.callThrough()
+      ws.socket.addEventListener 'open', () ->
+        ws.socket.close()
+      ws.socket.addEventListener 'close', () ->
+        setTimeout( ->
+          expect(ws.reconnect).toHaveBeenCalled()
+          jasmine.clock().uninstall()
+          ws.disconnect()
+          done()
+        , 30000)
+        jasmine.clock().tick(30010)
```

Comment:
actually you'll need to change strategy of the test here, because we are reconnecting once connection is closed:
```coffeescript
@socket.addEventListener 'close', =>
      
  if @options.reconnect
    @emit 'reconnecting'
    @reconnect(@options.reconnectionMaxTimeout)
```
you may better count calls for reconnect